### PR TITLE
refactor: stop duplicating `strip_channel_alias` from rattler

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Found while working on pixi global

### CODE REVIEW

File: src/cli/search.rs [hunk 1]
```diff
@@ -407,11 +409,10 @@ fn print_matching_packages<W: Write>(
...
         // https://github.com/mamba-org/rattler/issues/146
 
-        let channel_name = package
-            .channel
-            .strip_prefix(channel_config.channel_alias.as_str())
-            .unwrap_or(package.channel.as_str())
-            .trim_end_matches('/');
+        let channel_name = Url::from_str(&package.channel)
+            .ok()
+            .and_then(|url| channel_config.strip_channel_alias(&url))
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply src/cli/search.rs:414]: Unfortunately not, strip_channel_alias returns an `Option` not a `Result`

[the remark this answers is not in the document]
Comment (nichmor) [reply src/cli/search.rs:414]: but we call .ok() on Result of from_str , no?

[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply src/cli/search.rs:414]: I'd be happy if you'd make it work locally and push to my branch :)

[the remark this answers is not in the document]
Comment (nichmor) [reply src/cli/search.rs:414]: I get now what you mean , I think we can call just `unwrap_or`

[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply src/cli/search.rs:414]: As discussed on discord, `unwrap_or` would have an additional string allocation for the happy case

**Review by nichmor — APPROVED** [review]

**Pull request merged**